Build on s390x fails

## Expected Behavior

Builds

## Actual Behavior

Fails:
```
[  676s] [ 12%] Building CXX object src/CMakeFiles/hpx.dir/runtime/threads/coroutines/swapcontext.cpp.o
[  676s] cd /home/abuild/rpmbuild/BUILD/hpx-1.2.0-rc1/build/src && /usr/bin/c++  -DBOOST_DISABLE_ASSERTS -DHPX_COMPONENT_NAME=hpx -DHPX_DISABLE_ASSERTS -DHPX_ENABLE_ASSERT_HANDLER -DHPX_EXPORTS -D_GNU_SOURCE -Dhpx_EXPORTS -I/home/abuild/rpmbuild/BUILD/hpx-1.2.0-rc1 -I/home/abuild/rpmbuild/BUILD/hpx-1.2.0-rc1/build  -O2 -g -m64 -fmessage-length=0 -D_FORTIFY_SOURCE=2 -fstack-protector -funwind-tables -fasynchronous-unwind-tables -DNDEBUG -O2 -g -DNDEBUG -fPIC   -pthread -Wall -Wextra -Wno-unused-local-typedefs -Wno-strict-aliasing -Wno-sign-promo -Wno-attributes -Wno-cast-align -Wno-unused-parameter -Wformat=2 -Wno-format-nonliteral -Winit-self -Wdouble-promotion -Wcast-qual -Wcast-align -Werror=trampolines -Werror=parentheses -Werror=reorder -Werror=return-type -Werror=sequence-point -Werror=uninitialized -Werror=format -Werror=missing-braces -Werror=sign-compare -fdiagnostics-show-option -Werror=vla -Wno-unused-but-set-parameter -Wno-unused-but-set-variable -Wno-maybe-uninitialized -Wno-sync-nand -Wno-delete-non-virtual-dtor -fvisibility=hidden -std=c++17 -o CMakeFiles/hpx.dir/runtime/threads/coroutines/swapcontext.cpp.o -c /home/abuild/rpmbuild/BUILD/hpx-1.2.0-rc1/src/runtime/threads/coroutines/swapcontext.cpp
[  676s] /home/abuild/rpmbuild/BUILD/hpx-1.2.0-rc1/src/runtime/threads/coroutines/swapcontext.cpp:19:2: error: #error Unsupported platform
[  676s]  #error Unsupported platform
[  676s]   ^~~~~
[  676s] make[2]: *** [src/CMakeFiles/hpx.dir/build.make:1028: src/CMakeFiles/hpx.dir/runtime/threads/coroutines/swapcontext.cpp.o] Error 1
```

## Steps to Reproduce the Problem

Build on s390x, details [here](https://github.com/STEllAR-GROUP/hpx/files/2531946/build_s390x.txt)

## Specifications

  - HPX Version: 1.2.0-rc1
  - Platform (compiler, OS): gcc-8.2.1, OpenSuse

Ok with `-DHPX_WITH_GENERIC_CONTEXT_COROUTINES=ON` (as suggested in https://github.com/STEllAR-GROUP/hpx/issues/3512#issuecomment-434595715), it goes a bit further, but then fails with:
```
[  703s] [ 12%] Building CXX object src/CMakeFiles/hpx.dir/runtime/threads/detail/scheduled_thread_pool.cpp.o
[  703s] cd /home/abuild/rpmbuild/BUILD/hpx-1.2.0-rc1/build/src && /usr/bin/c++  -DBOOST_DISABLE_ASSERTS -DHPX_COMPONENT_NAME=hpx -DHPX_DISABLE_ASSERTS -DHPX_ENABLE_ASSERT_HANDLER -DHPX_EXPORTS -D_GNU_SOURCE -Dhpx_EXPORTS -I/home/abuild/rpmbuild/BUILD/hpx-1.2.0-rc1 -I/home/abuild/rpmbuild/BUILD/hpx-1.2.0-rc1/build  -O2 -g -m64 -fmessage-length=0 -D_FORTIFY_SOURCE=2 -fstack-protector -funwind-tables -fasynchronous-unwind-tables -DNDEBUG -O2 -g -DNDEBUG -fPIC   -pthread -Wall -Wextra -Wno-unused-local-typedefs -Wno-strict-aliasing -Wno-sign-promo -Wno-attributes -Wno-cast-align -Wno-unused-parameter -Wformat=2 -Wno-format-nonliteral -Winit-self -Wdouble-promotion -Wcast-qual -Wcast-align -Werror=trampolines -Werror=parentheses -Werror=reorder -Werror=return-type -Werror=sequence-point -Werror=uninitialized -Werror=format -Werror=missing-braces -Werror=sign-compare -fdiagnostics-show-option -Werror=vla -Wno-unused-but-set-parameter -Wno-unused-but-set-variable -Wno-maybe-uninitialized -Wno-sync-nand -Wno-delete-non-virtual-dtor -fvisibility=hidden -std=c++17 -o CMakeFiles/hpx.dir/runtime/threads/detail/scheduled_thread_pool.cpp.o -c /home/abuild/rpmbuild/BUILD/hpx-1.2.0-rc1/src/runtime/threads/detail/scheduled_thread_pool.cpp
[  704s] In file included from /home/abuild/rpmbuild/BUILD/hpx-1.2.0-rc1/hpx/runtime/threads/detail/scheduling_loop.hpp:18,
[  704s]                  from /home/abuild/rpmbuild/BUILD/hpx-1.2.0-rc1/hpx/runtime/threads/detail/scheduled_thread_pool_impl.hpp:21,
[  704s]                  from /home/abuild/rpmbuild/BUILD/hpx-1.2.0-rc1/src/runtime/threads/detail/scheduled_thread_pool.cpp:7:
[  704s] /home/abuild/rpmbuild/BUILD/hpx-1.2.0-rc1/hpx/util/hardware/timestamp.hpp:41:6: error: #error Unsupported platform.
[  704s]      #error Unsupported platform.
[  704s]       ^~~~~
[  711s] make[2]: *** [src/CMakeFiles/hpx.dir/build.make:1067: src/CMakeFiles/hpx.dir/runtime/threads/detail/scheduled_thread_pool.cpp.o] Error 1
```

Details [here](https://github.com/STEllAR-GROUP/hpx/files/2536225/s390x_log.txt)

--- a/hpx/runtime/threads/coroutines/detail/context_impl.hpp
+++ b/hpx/runtime/threads/coroutines/detail/context_impl.hpp
@@ -100,8 +100,8 @@
      make copyable.
 */
 
-#if defined(HPX_HAVE_GENERIC_CONTEXT_COROUTINES)                              \
- && defined(HPX_HAVE_FIBER_BASED_COROUTINES)
+#if defined(HPX_HAVE_GENERIC_CONTEXT_COROUTINES) &&                            \
+    defined(HPX_HAVE_FIBER_BASED_COROUTINES)
 #   error HPX_HAVE_GENERIC_CONTEXT_COROUTINES and HPX_HAVE_FIBER_BASED_COROUTINES cannot be defined at the same time.
 #endif
 
@@ -113,8 +113,8 @@ namespace hpx { namespace threads { namespace coroutines { namespace detail
     typedef generic_context::context_impl default_context_impl;
 }}}}
 
-#elif (defined(__linux) || defined(linux) || defined(__linux__)) \
-       && !defined(__bgq__) && !defined(__powerpc__)
+#elif (defined(__linux) || defined(linux) || defined(__linux__)) &&            \
+    !defined(__bgq__) && !defined(__powerpc__) && !defined(__s390x_)
 
 #include <hpx/runtime/threads/coroutines/detail/context_linux_x86.hpp>
 namespace hpx { namespace threads { namespace coroutines { namespace detail
@@ -122,7 +122,8 @@ namespace hpx { namespace threads { namespace coroutines { namespace detail
     typedef lx::context_impl default_context_impl;
 }}}}
 
-#elif defined(_POSIX_VERSION) || defined(__bgq__) || defined(__powerpc__)
+#elif defined(_POSIX_VERSION) || defined(__bgq__) || defined(__powerpc__) ||   \
+    defined(__s390x_)
 
 #include <hpx/runtime/threads/coroutines/detail/context_posix.hpp>
 namespace hpx { namespace threads { namespace coroutines { namespace detail

--- a/hpx/util/hardware/timestamp.hpp
+++ b/hpx/util/hardware/timestamp.hpp
@@ -35,6 +35,8 @@
     #include <hpx/util/hardware/timestamp/linux_generic.hpp>
 #elif defined(__powerpc__)
     #include <hpx/util/hardware/timestamp/linux_generic.hpp>
+#elif defined(__s390x__)
+    #include <hpx/util/hardware/timestamp/linux_generic.hpp>
 #elif defined(__bgq__)
     #include <hpx/util/hardware/timestamp/bgq.hpp>
 #else

--- a/src/runtime/threads/coroutines/swapcontext.cpp
+++ b/src/runtime/threads/coroutines/swapcontext.cpp
@@ -8,8 +8,8 @@
 
 #if !defined(HPX_HAVE_GENERIC_CONTEXT_COROUTINES)
 
-#if (defined(__linux) || defined(linux) || defined(__linux__)) \
- && !defined(__bgq__) && !defined(__powerpc__)
+#if (defined(__linux) || defined(linux) || defined(__linux__)) &&              \
+    !defined(__bgq__) && !defined(__powerpc__) && !defined(__s390x__)
 
 #if defined(__x86_64__) || defined(__amd64__)
 #include "swapcontext64.ipp"

